### PR TITLE
AVM2 Primitive Class Methods & Properties

### PR DESCRIPTION
**Pull request by kmeisthax**
This implements the rest of `Number`, `Boolean`, `int`, and `uint`. The previous PR only contained constructors.

- [x] `Number`
  - [x] Constants
  - [x] `Number.toExponential`
  - [x] `Number.toFixed`
  - [x] `Number.toPrecision`
  - [x] `Number.toString`
  - [x] `Number.valueOf`
- [x] Boolean
  - [x] `Boolean.toString`
  - [x] `Boolean.valueOf`
- [x] `int`
  - [x] Constants
  - [x] `int.toExponential`
  - [x] `int.toFixed`
  - [x] `int.toPrecision`
  - [x] `int.toString`
  - [x] `int.valueOf`
- [x] uint
  - [x] Constants
  - [x] `uint.toExponential`
  - [x] `uint.toFixed`
  - [x] `uint.toPrecision`
  - [x] `uint.toString`
  - [x] `uint.valueOf`

### CODE REVIEW

File: core/src/avm2/globals/boolean.rs [hunk 1]
```diff
@@ -53,6 +53,39 @@ pub fn class_init<'gc>(
...
 
+/// Implements `Boolean.toString`
+pub fn to_string<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Some(this) = this.as_primitive() {
+            match this.coerce_to_boolean() {
```
Comment (adrian17) [inline core/src/avm2/globals/boolean.rs:64]: FYI, in FP, I'm getting:
```js
var a = Boolean.prototype.toString.apply(1);
```
```
TypeError: Error #1004: Method Boolean.prototype.toString was invoked on an incompatible object.
```
(Not a big immediate deal if you think this should be refactored more widely)

Comment (kmeisthax) [reply core/src/avm2/globals/boolean.rs:64]: The documentation only lists these methods in the `AS3` namespace; I didn't include them in `public`. However, upon further tests it appears overridable `public` versions of these methods exist in on these classes, so I'm going to implement those as well. As part of that I'll also make that particular method throw an error.

File: core/src/avm2/globals/int.rs [hunk 2]
```diff
@@ -46,10 +47,214 @@ pub fn native_instance_init<'gc>(
...
+            )
+            .into(),
+        )?;
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `int.toExponential`
+pub fn to_exponential<'gc>(
```
Comment (adrian17) [inline core/src/avm2/globals/int.rs:128]: Let's make new methods non-`pub` by default? :D

File: core/src/avm2/globals/boolean.rs [hunk 3]
```diff
@@ -46,10 +46,78 @@ pub fn native_instance_init<'gc>(
...
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        let mut boolean_proto = this
+            .get_property(this, &QName::dynamic_name("prototype").into(), activation)?
+            .coerce_to_object(activation)?;
+        let scope = activation.create_scopechain();
+        let gc_context = activation.context.gc_context;
+        let this_class = this.as_class_object().unwrap();
+
+        boolean_proto.install_dynamic_property(
```
Comment (adrian17) [inline core/src/avm2/globals/boolean.rs:61]: (note: I'm getting tempted to attempt making method-on-prototype declarations declarative, like normal class methods)

**Review by adrian17 — APPROVED** [review]

**Pull request merged**